examples/gno.land/p/demo/grc/grc20/types.gno

```diff
@@ -12,9 +13,24 @@ var (
 	ErrInsufficientAllowance = errors.New("insufficient allowance")
 	ErrInvalidAddress        = errors.New("invalid address")
 	ErrCannotTransferToSelf  = errors.New("cannot send transfer to self")
+	ErrReadonly              = errors.New("banker is readonly")
+	ErrRestrictedBankOwner   = errors.New("restricted to bank owner")
 )
 
-type Token interface {
+type Bank struct {
```

Comment:
Also, what do you think of making it compatible with the `GRC20` interface, but panicking on any write method?

Or maybe you already wanted this with the MockBanker?

Reply:
The read-only part is compatible in both signature and usage.

The write part shouldn't cause panic; the signature actually differs. A bank can impersonate any owner, so all the write helpers include an additional "from" argument.

For this reason: 1. It's not possible to be fully GRC20 compatible; 2. It will create unrealistic expectations that the bank should be used for this, in my opinion.